cardinal/ecs/guid.go

```diff
@@ -0,0 +1,81 @@
+package ecs
+
+import (
+	"errors"
+
+	"github.com/cometbft/cometbft/crypto"
+)
+
+const numOfCharactersInGUID = 24
+
+// GUIDComponent is a unique game identification string. It is generated randomly on tick 0 and
+// will never change.
+type GUIDComponent struct {
+	GUID string `json:"guid"`
+}
+
+func (GUIDComponent) Name() string {
+	return "GameUniqueIDComponent"
+}
+
+var ErrGUIDNotFound = errors.New("guid not found")
+
+func getGUID(wCtx WorldContext) (guid string, err error) {
+	search, err := wCtx.NewSearch(Exact(GUIDComponent{}))
+	if err != nil {
+		return "", err
+	}
+	count, err := search.Count(wCtx)
+	if err != nil {
+		return "", err
+	}
+	if count == 0 {
+		return "", ErrGUIDNotFound
+	}
+	id, err := search.First(wCtx)
+	if err != nil {
+		return "", err
+	}
+	guidComp, err := GetComponent[GUIDComponent](wCtx, id)
+	if err != nil {
+		return "", err
+	}
+	return guidComp.GUID, nil
+}
+
+func CreateGUIDSystem(wCtx WorldContext) error {
```

Comment:
it would be nice if we could figure out a way later down the line to have these singleton style components not have to be a full on system to get their data injected into the ECS storage. Maybe we can make an initializer type of thing.